components/announcement_bar/cloud_delinquency/index.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import {useHistory} from 'react-router-dom';
+
+import {t} from 'utils/i18n';
+import {
+    AnnouncementBarTypes,
+} from 'utils/constants';
+
+import {GlobalState} from 'types/store';
+import AnnouncementBar from '../default_announcement_bar';
+import useGetSubscription from 'components/common/hooks/useGetSubscription';
+import {isSystemAdmin} from 'mattermost-redux/utils/user_utils';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+
+const CloudDelinquencyAnnouncementBar: React.FC = () => {
+    const subscription = useGetSubscription();
+    const history = useHistory();
+
+    const currentUser = useSelector((state: GlobalState) =>
+        getCurrentUser(state),
+    );
+
+    const shouldShowBanner = () => {
+        if (!subscription) {
+            return false;
+        }
+
+        if (!isSystemAdmin(currentUser.roles)) {
+            return false;
+        }
+
+        if (!subscription.delinquent_since) {
+            return false;
+        }
+
+        return true;
+    };
+
+    const getBannerType = () => {
+        const delinquencyDate = new Date(
+            (subscription?.delinquent_since || 0) * 1000,
+        );
+
+        const oneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds
+        const today = new Date();
+        const diffDays = Math.round(
+            Math.abs((today.valueOf() - delinquencyDate.valueOf()) / oneDay),
+        );
+        if (diffDays > 90) {
```

Comment:
Super nit: from the CWS side, the calculation is `>= 90`). Do we need to round at all here, and should it be `>= 90` to match CWS more closely (we're talking the difference of half a day here, since would need to be 90 in CWS, vs 90.5 here to round up to 91).

Reply:
I did do this intentionally to allow time for the job to run on the CWS side and perform the downgrade before the colour would switch to red. I don't think it's a huge deal either way, 0/5 on if we should be inclusive or not